harvester_e2e_tests/integration/test_networks.py

```diff
@@ -0,0 +1,840 @@
+            ip_addresses = interface['ipAddresses']
+
+        # # Ping management ip address
+        mgmt_ip = ip_addresses[0]
+        ping_command = "ping -c 50 {0}".format(mgmt_ip)
+
+        _stdout, _stderr = self.ssh_client(
+            client, client_ip, node_user, node_password, ping_command, wait_timeout)
+
+        stdout = _stdout.read().decode('ascii').strip("\n")
+
+        assert stdout.find("64 bytes from {0}".format(mgmt_ip)) > 0, (
+            'Failed to ping VM management IP %s' % (mgmt_ip))
+
+        # SSH to management ip address and execute command
+        _stdout, _stderr = self.ssh_jumpstart(
+            client, mgmt_ip, client_ip, node_user, node_password,
+            vm_credential["user"], vm_credential["password"], "ls")
+
+        stdout = _stdout.read().decode('ascii').strip("\n")
+
+        assert stdout.find("bin") == 0, (
+            'Failed to ssh to VM management IP %s' % (mgmt_ip))
+
+        # cleanup VM
+        delete_vm(api_client, unique_name, wait_timeout)
+
+    @pytest.mark.p0
+    @pytest.mark.dependency(name="vlan_network_connection",
+                            depends=["mgmt_network_connection"])
+    def test_vlan_network_connection(self, api_client, request, client, unique_name,
+                                     vlan_network, vlan_subnet_prefix_store):
+        """
+        Manual test plan reference:
+        https://harvester.github.io/tests/manual/network/validate-network-external-vlan/
+
+
+        Steps:
+        1. Create an external VLAN network
+        2. Create a new VM already with the vlan network
+        3. Check can ping external VLAN IP
+        4. Check can SSH to VM from external IP
+        """
+        wait_timeout = request.config.getoption('--wait-timeout')
+        unique_name = unique_name + "-vlan"
+
+        # client_ip = request.config.getoption('--endpoint').strip('https://')
+
+        spec = api_client.vms.Spec(1, 2, mgmt_network=False)
+        spec.user_data += cloud_user_data.format(password=vm_credential["password"])
+
+        # Create VM
+        spec.add_image(image_name, "default/" + image_name)
+        spec.add_network("default", vlan_network['id'])
+
+        code, data = api_client.vms.create(unique_name, spec)
+        assert 201 == code, (f"Failed to create vm with error: {code}, {data}")
+
+        # Check VM start in running state
+        check_vm_running(api_client, unique_name, wait_timeout)
+
+        # Check until VM ip address exists
+        check_vm_ip_exists(api_client, unique_name, wait_timeout)
+
+        # Get VM interface ipAddresses
+        code, data = api_client.vms.vm_instance(unique_name)
+        interfaces_data = data['status']['interfaces']
+        for interface in interfaces_data:
+            ip_addresses = interface['ipAddresses']
+
+        vlan_ip = ip_addresses[0]
+
+        vlan_subnet_prefix = vlan_ip.rsplit('.', 1)[0]
+        vlan_subnet_prefix_store["value"] = vlan_subnet_prefix
```

Comment:
Appreciated for the review. 
The purpose to store the vlan_subnet information is we need to use the vlan subnet information e.g `10.84.99` to identify when we switch from mgmt to vlan network in test case 5: `test_mgmt_to_vlan_connection`

This can ensure the exposed ip address 10.84.99.x did actually match the subnet information. 
Thus we could determine it actually switch from management subnet (10.52) to vlan subnet (10.84.99) accordingly. 

Without this information, we need extra code effort to access the the gitlab repository 
`harvester-baremetal-info/-/blob/master/pcloud001/dhcp.yml` to retrieve the `vm_subnet` field

Reply:
What if the mask number is not fully divided by eight?
When you use `"192.168.10.1".rsplit(".", 1)[0]` to get the **vlan_subnet_prefix**, which means the prefix is `192.168.10`, but do we guarantee that the CIDR is `192.168.10.0/24`?
What if CIDR is `192.168.10.0/16` or even `/15`, `/25`, `/26`?

And according to
https://github.com/harvester/tests/pull/681/files#diff-e30c67d488529228b4fa2f07596daadf626451f5114bc2a6c8622c4e25e87f04R504
when you use
```python
if vlan_subnet_prefix in ip_addresses[0]: ...
```
consider the case:
```python
vlan_subnet_prefix = '10.52.1'
ip_addresses = ["10.10.52.1"]
```
Is the case should be *True* in the if condition?


> Without this information, we need extra code effort to access the the gitlab repository
harvester-baremetal-info/-/blob/master/pcloud001/dhcp.yml to retrieve the vm_subnet field

We didn't declare the mgmt network should always be `10.52.x.x/y`, so why we have to check it by its ip address? (and the test plan didn't mentioned that?)
When you look into the manual test plan, **Expected Results** described clearly: Check the VM is able to access from external network, but nothing about its IP address.

If we are  going to check the **interface** comes from which NIC, the name field is more useful, consider the schema:
```python
 'interfaces': [{'infoSource': 'domain, guest-agent',
                 'interfaceName': 'enp1s0',
                 'ipAddress': '10.52.1.23',
                 'ipAddresses': ['10.52.1.23'],
                 'mac': '52:54:00:8d:1f:56',
                 'name': 'default'}]
```
the `name` field is reflect to the name you defined for the NIC, when we adding a NIC to a VM, we are always give it a name, so we can simply check interface's name to know it is from which NIC.